ptf/sairif.py

```diff
@@ -0,0 +1,7955 @@
+        self.port13_rif_counter_out = 0
+        self.lag1_rif_counter_in = 0
+        self.lag1_rif_counter_out = 0
+
+        dmac1 = '00:11:22:33:44:55'
+        dmac3 = '00:33:22:33:44:55'
+
+        self.lag1_rif = sai_thrift_create_router_interface(
+            self.client,
+            type=SAI_ROUTER_INTERFACE_TYPE_PORT,
+            virtual_router_id=self.default_vrf,
+            port_id=self.lag1)
+
+        self.nhop1 = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress('10.10.10.2'),
+            router_interface_id=self.port10_rif,
+            type=SAI_NEXT_HOP_TYPE_IP)
+        self.neighbor_entry1 = sai_thrift_neighbor_entry_t(
+            rif_id=self.port10_rif, ip_address=sai_ipaddress('10.10.10.2'))
+        sai_thrift_create_neighbor_entry(
+            self.client, self.neighbor_entry1, dst_mac_address=dmac1)
+
+        self.neighbor_entry3 = sai_thrift_neighbor_entry_t(
+            rif_id=self.lag1_rif, ip_address=sai_ipaddress('12.10.10.2'))
+        sai_thrift_create_neighbor_entry(
+            self.client, self.neighbor_entry3, dst_mac_address=dmac3)
+        self.nhop3 = sai_thrift_create_next_hop(
+            self.client,
+            ip=sai_ipaddress('12.10.10.2'),
+            router_interface_id=self.lag1_rif,
+            type=SAI_NEXT_HOP_TYPE_IP)
+
+        self.route_entry0 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('10.10.10.1/32'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_entry0, next_hop_id=self.nhop1)
+
+        self.route_entry0_lpm = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('11.11.11.0/24'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_entry0_lpm, next_hop_id=self.nhop1)
+
+        self.route_entry1 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(
+                '1234:5678:9abc:def0:4422:1133:5577:99aa/128'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_entry1, next_hop_id=self.nhop1)
+
+        self.route_entry1_lpm = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('4000::0/65'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_entry1_lpm, next_hop_id=self.nhop1)
+
+        self.route_lag0 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('12.10.10.1/32'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_lag0, next_hop_id=self.nhop3)
+
+        self.route_lag1 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(
+                '1234:5678:9abc:def0:1122:3344:5566:7788/128'))
+        sai_thrift_create_route_entry(
+            self.client, self.route_lag1, next_hop_id=self.nhop3)
+
+    def runTest(self):
```

Comment:
The assumption was to divide router interface tests into 3 classes - L3 interface, Sub-port and SVI and under each class have tests with common configuration. In my opinion I would leave it as it is now because it's more clear to understand the nature of these tests, but I could change it if you need smaller classes, but I'm not sure if it wouldn't just duplicate the config code.

Reply:
try to resolve this comment from the PTF stcuture
https://github.com/p4lang/ptf/pull/168